### ISSUE TITLE
`Style/MutableConstant` gives false positive on splat assignment

### ISSUE THREAD

**Issue by Drenmi** [bbatsov/rubocop#3366]
As per [this StackOverflow post](http://stackoverflow.com/questions/38757237/ruby-rubocop-how-to-freeze-an-array-constant-generated-with-splat), when running this cop on a piece of code with a splat assignment, it incorrectly reports a violation.

Example:

```
FOO = *(1...10)
```

## Expected behavior

No offenses are registered.

## Actual behavior

```
C: Freeze mutable objects assigned to constants.
NUMS = *(1...10)
       ^^^^^^^^^
```

## Steps to reproduce the problem

Run:

```
bundle exec rubocop --only Style/MutableConstant
```

on the above code.

## RuboCop version

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.0 x86_64-darwin15)
```
